docs/websites/lamp/lamp-server-on-ubuntu-14-04-trusty.md

```diff
@@ -0,0 +1,298 @@
+
+1. Ubuntu includes packages for installing PHP5 with support for PHP5-FPM. From the terminal, issue the following command:
+
+        sudo apt install libapache2-mod-fastcgi php5-fpm php5
+
+2. We will now configure Apache to pass all requests for PHP files, with the _php_ file extension, to the PHP wrapper through FastCGI. Once PHP5 is installed we'll want to ensure that the PHP5-FPM module is enabled. Issue the following command to do this:
+
+        sudo a2enmod actions fastcgi alias
+        sudo a2enconf php5-fpm
+
+3. It is best to configure PHP-FPM to use UNIX sockets instead of TCP. To verify if this is already the case, issue the following command in your terminal:
+
+        sudo grep -E '^\s*listen\s*=\s*[a-zA-Z/]+' /etc/php5/fpm/pool.d/www.conf
+
+    You should see the following output:
+
+        listen = /var/run/php5-fpm.sock
+
+    If you see the above output, skip to step 6.
+
+3. If no output is returned, you will need to edit the following file and add this line:
+
+    {: .file-excerpt } 
+    etc/php5/fpm/pool.d/www.conf
+    :   ~~~
+        listen = /var/run/php5-fpm.sock
+        ~~~
+
+4. Find the following line and comment it out, if it is not already.
+
+    {: .file-excerpt }
+    /etc/php5/fpm/pool.d/www.conf
+    :   ~~~
+        listen = 127.0.0.1:9000
+        ~~~
+
+5. Restart the php5-fpm daemon for these changes to take effect.
+
+        sudo service php5-fpm restart
+
+6. Check for the version of Apache with the following command.
+
+        apache2 -v
+
+7. We will need to modify the file `fastcgi.conf`. As with all default configuration files, it is a good idea to back them up before modifying them in case you need to start over. Backup `fastcg.conf` by entering the following command:
+
+        sudo cp /etc/apache2/mods-enabled/fastcgi.conf ~/fastcgi.conf.backup
+
+8. Depending on your Apache version, edit the following file accordingly.
+
+    **Apache 2.2 or earlier**
+
+    {: .file-excerpt }
+    /etc/apache2/mods-enabled/fastcgi.conf
+    :   ~~~
+        <IfModule mod_fastcgi.c>
+         AddType application/x-httpd-fastphp5 .php
+         Action application/x-httpd-fastphp5 /php5-fcgi
+         Alias /php5-fcgi /usr/lib/cgi-bin/php5-fcgi
+         FastCgiExternalServer /usr/lib/cgi-bin/php5-fcgi -socket /var/run/php5-fpm.sock -pass-header Authorization
+        </IfModule>
+        ~~~
+
+    **Apache 2.4 or later**
+
+    {: .file-excerpt }
+    /etc/apache2/mods-enabled/fastcgi.conf
+    :   ~~~
+        <IfModule mod_fastcgi.c>
+         AddType application/x-httpd-fastphp5 .php
+         Action application/x-httpd-fastphp5 /php5-fcgi
+         Alias /php5-fcgi /usr/lib/cgi-bin/php5-fcgi
+         FastCgiExternalServer /usr/lib/cgi-bin/php5-fcgi -socket /var/run/php5-fpm.sock -pass-header Authorization
```

Comment:
It's fairly amusing that, because you're using `FastCgiExternalServer` which doesn't actually care about starting anything, `/usr/lib/cgi-bin/php5-fcgi` doesn't need to actually exist, yet you need to have *something* there to be able to make your `Action` routing work.